src/ast/passes/codegen_llvm.cpp

```diff
@@ -2690,7 +2712,8 @@ void CodegenLLVM::createTupleCopy(const SizedType &expr_type,
                               { b_.getInt32(0), b_.getInt32(i) });
     if (t_type.IsTupleTy() && !t_type.IsSameSizeRecursive(var_type)) {
       createTupleCopy(t_type, var_type.GetField(i).type, dst, offset_val);
-    } else if (t_type.IsIntTy() && t_type.GetSize() < 8) {
+    } else if ((t_type.IsIntTy() || t_type.IsBoolTy()) &&
+               t_type.GetSize() < 8) {
```

Comment:
I think this check is wrong - it doesn't match the corresponding check on line 3446, which stores bools as 8-bits, not 64-bits.

Here's a program that (I think) fails verification because of this:
```
@mmm[true, true, 123] = 1111; @mmm[true, true, (int32)456] = 2222;
```